src/docfx/SubCommands/MetadataMerger.cs

```diff
@@ -104,6 +111,103 @@ private static void MergeToc(MetadataMergeParameters parameters, string outputBa
                 YamlMime.TableOfContent);
         }
 
+        private void InitMetaTable(FileModel model)
+        {
+            var content = model.Content as PageViewModel;
+            if (content != null && model.Uids.Length > 0)
+            {
+                _metaTable.Add(model.Uids.First().Name, content.Metadata);
+            }
+        }
+
+        private void SetTocViewModelMetadata(TocViewModel vm, ImmutableList<string> metaNames)
+        {
+            foreach (var item in vm)
+            {
+                foreach (var metaName in metaNames)
+                {
+                    MergeIntoTocMetadata(item, metaName);
+                }
+            }
+        }
+
+        private List<string> MergeIntoTocMetadata(TocItemViewModel item, string metaName)
+        {
+            Dictionary<string, object> metadata;
+            if (_metaTable.TryGetValue(item.Uid, out metadata))
+            {
+                object metaValue;
+                if (metadata.TryGetValue(metaName, out metaValue))
+                {
+                    var merged = TryGetListFromObject(metaValue);
+                    foreach (var child in item.Items ?? Enumerable.Empty<TocItemViewModel>())
+                    {
+                        var childMetaValue = MergeIntoTocMetadata(child, metaName);
+                        merged = MergeMetadata(merged, childMetaValue);
+                    }
+                    if (item.Metadata == null)
+                    {
+                        item.Metadata = new Dictionary<string, object>();
+                    }
+                    item.Metadata[metaName] = merged;
+                    return merged;
+                }
+            }
+            return null;
+        }
+
+        private static List<string> MergeMetadata(object meta1, object meta2)
+        {
+            var list1 = TryGetListFromObject(meta1);
```

Comment:
it could be a string or a list of string.

---
In reply to: [90794622](https://github.com/dotnet/docfx/pull/941#discussion_r90794622) [](ancestors = 90794622)